cloudify_gcp/utils.py

```diff
@@ -0,0 +1,466 @@
+
+    @check_response
+    def get(self):
+        self.last_response = self._get()
+        self.last_status = self.last_response['status']
+        return self.last_response
+
+    @abstractmethod
+    def _get(self): pass
+
+
+class GlobalOperation(Operation):
+    def _get(self):
+        return self.discovery.globalOperations().get(
+            project=self.project,
+            operation=self.name).execute()
+
+
+class RegionOperation(Operation):
+    def _get(self):
+        return self.discovery.regionOperations().get(
+            project=self.project,
+            region=basename(self.region),
+            operation=self.name).execute()
+
+
+class ZoneOperation(Operation):
+    def _get(self):
+        return self.discovery.zoneOperations().get(
+            project=self.project,
+            zone=basename(self.zone),
+            operation=self.name).execute()
+
+
+def get_relationships(
+        relationships,
+        filter_relationships=None,
+        filter_nodes=None):
+    """
+    Get all relationships of a particular node or the current context.
+
+    Optionally filter based on relationship type, node type.
+    """
+    if isinstance(relationships, (CloudifyContext, Proxy)):
+        # Shortcut to support supplying ctx directly
+        relationships = relationships.instance.relationships
+    # And coerce the other inputs to lists if they are strings:
+    if isinstance(filter_relationships, basestring):
+        filter_relationships = [filter_relationships]
+    if isinstance(filter_nodes, basestring):
+        filter_nodes = [filter_nodes]
+    results = []
+    for rel in relationships:
+        if filter_relationships and rel.type not in filter_relationships:
+            rel = None
+        if filter_nodes and rel.target.node.type not in filter_nodes:
+            rel = None
+        if rel:
+            results.append(rel)
+    return results
+
+
+def get_network_node(ctx):
+    network_list = get_relationships(
+            ctx,
+            filter_relationships='cloudify.gcp.relationships.'
+                                 'contained_in_network',
+            )
+    if len(network_list) > 0:
+        return network_list[0].target
+
+
+def get_net_and_subnet(ctx):
+    """
+    Returns a tuple of the ctx node's
+    (Network, Subnetwork)
+    """
+    net_node = get_network_node(ctx)
+
+    subnetwork = None
+    if net_node:
+        if net_node.node.type == 'cloudify.gcp.nodes.Network':
+            network = net_node.instance.runtime_properties['selfLink']
+        elif net_node.node.type == 'cloudify.gcp.nodes.SubNetwork':
+            network = net_node.instance.runtime_properties['network']
+            subnetwork = net_node.instance.runtime_properties['selfLink']
+        else:
+            raise NonRecoverableError(
+                'Unsupported target type for '
+                "'cloudify.gcp.relationships.instance_contained_in_network")
+    else:
+        network = get_gcp_config()['network']
```

Comment:
Idan also pointed out it's better for blueprints to be explicit rather than relying on defaults in config.
I think removing that everywhere is maybe too big a change for right now. Perhaps it could be hard-coded to `default` or the manager's network if running on a manager now, rather than removed completely?
(otherwise I can just document it, the GCP config file needs documenting anyway)